Try build/test different geese in CI

evalexpr opened the pull request:
First pass at #37 without support for cross-compilation of tests

Signed-off-by: evalexpr <wilkinsphysics@gmail.com>

bradfitz commented:
Thanks. I guess this should wait until there's a GOOS besides Linux that passes.

evalexpr commented:
@bradfitz Darwin seems to pass at least, I'll keep this open in the meantime.

I'm planning on writing the shell wrapper for the cross-compilation of tests, I'll push it up in a sec, but it again fails on Windows due to the same error.

Would appreciate some thoughts on it - also not sure if we want to split out the actions a bit, or are we happy to have them all as part of linux.yml?

danderson commented:
Well, good news, the test failure is legit! 7e5e3277 changed an API on some platforms only.

This is exactly what I wanted to catch by adding this to CI, there's a thinko in the darwin build that makes it straight up not compile.

I've fixed the darwin/windows/openbsd builds at HEAD (in theory), can you rebase and let CI run again?

Also, your second commit is missing a Signed-Off-By, which is making DCO bot sad.

evalexpr commented:
Rebased. Looks good now on Darwin but Windows is still failing due to #42  - I think after that is fixed we'll be good to go.

Meanwhile, I've added openbsd to the test script too - passes locally for me, but doesn't get a chance to run in the action due to the earlier failure when building for Windows.

I'll add back the Slack integration in the action after #42 is fixed.

bradfitz commented on .github/workflows/linux.yml:
```diff
@@ -28,20 +28,21 @@ jobs:
+        GOARCH: amd64
+      run: go build ./cmd/...
 
-    - uses: k0kubun/action-slack@v2.0.0
```
Why is this being deleted?

danderson replied:
Temporary, to not spam our slack.

#42 should be fixed now, so a rebase + rerun should do the trick.

evalexpr replied:
Looks green all around now. I've added this back in.

danderson commented:
\o/ all green. Thank you!
